crowbar_framework/app/models/ceph_service.rb

```diff
@@ -212,6 +212,19 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
         master.save
       end
     end
+
+    # estimating number of osds for ceph cluster
+    disks_count = 0
+    unless osd_nodes.empty? || role.default_attributes["ceph"]["config"]["osds_in_total"] != 0
+      osds = osd_nodes.map {|n| NodeObject.find_node_by_name n}
+      osds.each do |osd|
+        disks_count += osd.unclaimed_physical_drives.length
```

Comment:
You're assuming that ceph is configured to take all disks, while it can be set to take only the first one that is free.